File: 프로그래머스/1/12943. 콜라츠 추측/콜라츠 추측.js
function solution(num) { 
    let ans = 0;
    
    while (ans <= 500){
        if (num == 1) 
            return ans;
        
        if (num%2) {
            num = (num*3 + 1) / 2;
            ans += 2;
        } else {
            num /= 2;
            ans += 1;
        } 
    }
    
    return -1;
}

File: 프로그래머스/1/12943. 콜라츠 추측/콜라츠 추측.py
def solution(num):
    cnt = 0
    
    while(cnt <= 500):
        if num == 1:
            return cnt
        
        cnt += 1
        num, r = divmod(num, 2)
        
        if r:
            cnt += 1
            num = num*3 + 2
        
    return -1